sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient_private.cs

```diff
@@ -50,41 +50,12 @@ private static void ParseConnectionString(string connectionString, out Uri uri,
         {
             Debug.Assert(connectionString != null); // callers check this
 
-            uri = default;
-            credential = default;
-            secret = default;
+            var parsed = ConnectionString.Parse(connectionString, allowWhitespaces: true);
 
-            // Parse connection string
-            string[] args = connectionString.Split(new char[] { ';' }, StringSplitOptions.RemoveEmptyEntries);
-            if (args.Length < 3)
-            {
-                throw new ArgumentException("invalid connection string segment count", nameof(connectionString));
-            }
-
-            const string endpointString = "Endpoint=";
-            const string idString = "Id=";
-            const string secretString = "Secret=";
-
-            // TODO (pri 2): this allows elements in the connection string to be in varied order. Should we disallow it?
-            // TODO (pri 2): this parser will succeed even if one of the elements is missing (e.g. if cs == "a;b;c". We should fix that.
-            foreach (var arg in args)
-            {
-                var segment = arg.Trim();
-                if (segment.StartsWith(endpointString, StringComparison.OrdinalIgnoreCase))
-                {
-                    uri = new Uri(segment.Substring(segment.IndexOf('=') + 1));
-                }
-                else if (segment.StartsWith(idString, StringComparison.OrdinalIgnoreCase))
-                {
-                    credential = segment.Substring(segment.IndexOf('=') + 1);
-                }
-                else if (segment.StartsWith(secretString, StringComparison.OrdinalIgnoreCase))
-                {
-                    var secretBase64 = segment.Substring(segment.IndexOf('=') + 1);
-                    // TODO (pri 2): this might throw an obscure exception. Should we throw a consisten exception when the parser fails?
-                    secret = Convert.FromBase64String(secretBase64);
-                }
-            };
+            uri = new Uri(parsed.GetRequired("Endpoint"));
+            credential = parsed.GetRequired("Id");
+            // TODO (pri 2): this might throw an obscure exception. Should we throw a consistent exception when the parser fails?
```

Comment:
Let's take this comment out.  The fix should address it, or not, but changing the exception would be a breaking change that we'd like to avoid unless there's a compelling business justification for it.

Reply:
Fix doesn't address it cause the problem happens when base64 decode fails. I can wrap it with exception of the same type, but different message.

Reply:
Let's create an issue to track it then.  Will fixing it be a breaking change?

Reply:
`Convert.FromBase64String` throws `FormatException`. If you want to wrap it wih `InvalidOperationException`, then I think it could be a breaking change.

Reply:
Just to make sure we're both talking about the same thing ....  As long as the library is in preview, breaking changes are fine.  Once we GA the library, breaking changes are not allowed.  Keeping this comment in the source code implies that we have a TODO that would introduce a breaking change after GA.  We should either make a decision to do this now, or decide not to and remove the TODO.  It would be worth following the precedent set in other libraries.  Does storage use FromBase64String in their connection string parsing?  If so, do they wrap FormatException in InvalidOperationException?  My recommendation would be to follow their precedent to have consistent behavior across the Azure SDK.

Reply:
Storage checks if secret key is valid base64 string, and uses `null` if it isn't. We can't do this cause we don't allow `null` values for the secret. But I think the closest would be wrapping the exception with `InvalidOperationException`.

Reply:
Wrapped with `InvalidOperationException`